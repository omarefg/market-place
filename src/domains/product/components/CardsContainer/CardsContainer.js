import React from "react";
import classes from "./CardsContainer.module.css";

export function CardsContainer(props) {
  const { children, title } = props;
  return (
    <>
      <h2 data-testid="CardsContainer__title">{title}</h2>
      <div className={classes.cardsContainer}>{children}</div>
    </>
  );
}
