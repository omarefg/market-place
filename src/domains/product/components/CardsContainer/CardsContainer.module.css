.cardsContainer {
  font-size: calc(10px + 2vmin);
  color: var(--color-3);
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(192px, 1fr));
  gap: 50px;
  padding: 40px;
  max-width: 960px;
  margin: 0 auto;
}
