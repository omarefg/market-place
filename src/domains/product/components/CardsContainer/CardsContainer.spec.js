import { render } from "@testing-library/react";
import React from "react";
import { CardsContainer } from "./CardsContainer";

describe("@domains/product/components/CardsContainer", () => {
  let defaultProps, wrapper;

  beforeEach(() => {
    defaultProps = {
      children: <div data-testid="CardsContainer__children" />,
      title: "titleMock",
    };
    wrapper = (props = {}) => {
      const finalProps = { ...defaultProps, ...props };
      return render(<CardsContainer {...finalProps} />);
    };
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it("Given a valid call it must render without crashing", () => {
    // Arrange
    const { container } = wrapper();
    // Assert
    expect(container).toMatchSnapshot();
  });

  it("Given a valid call it must render the `title` prop", () => {
    // Arrange
    const { getByTestId } = wrapper();
    const titleContainer = getByTestId("CardsContainer__title");
    // Assert
    expect(titleContainer.innerHTML).toBe(defaultProps.title);
  });

  it("Given a valid call it must render `children` prop", () => {
    // Arrange
    const { container, getByTestId } = wrapper();
    // Act
    const children = getByTestId('CardsContainer__children');
    //Assert
    expect(container).toContainElement(children);
  });
});
