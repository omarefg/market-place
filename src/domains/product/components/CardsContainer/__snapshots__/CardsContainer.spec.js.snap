// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@domains/product/components/CardsContainer Given a valid call it must render without crashing 1`] = `
<div>
  <h2
    data-testid="CardsContainer__title"
  >
    titleMock
  </h2>
  <div
    class="cardsContainer"
  >
    <div
      data-testid="CardsContainer__children"
    />
  </div>
</div>
`;
